homeassistant/components/electric_kiwi/config_flow.py

```diff
@@ -45,7 +48,19 @@ async def async_step_reauth_confirm(
 
     async def async_oauth_create_entry(self, data: dict) -> ConfigFlowResult:
         """Create an entry for Electric Kiwi."""
-        existing_entry = await self.async_set_unique_id(DOMAIN)
+
+        ek_api = ElectricKiwiApi(
+            api.ConfigFlowElectricKiwiAuth(self.hass, data["token"]["access_token"])
+        )
+
+        session = await ek_api.get_active_session()
+
+        if len(session.customer_numbers) == 0:
+            return self.async_abort(reason="no_customers")
+
+        unique_id = "_".join(str(num) for num in session.customer_numbers)
+        existing_entry = await self.async_set_unique_id(unique_id)
```

Comment:
if we have an unique id, we should use this id to also abort if it is already setup, and when reauthentiucating we should make sure that the unique id is the same

Reply:
abort if configured added

Reply:
But what about reauth? Please check an integration like withings for that

Reply:
done

Reply:
You missed the `self._abort_if_unique_id_mismatch(reason="wrong_account")` where someone reauths with a completely different account